File: 492-ConstructTheRectangle/492-ConstructTheRectangle.py
# Last updated: 7/31/2025, 12:29:25 PM
class Solution:
    def constructRectangle(self, area: int) -> List[int]:
        b=int(area**(1/2))
        while(area%b!=0):
            b-=1
        l=int(area/b)
        if(b==1):
            return [area,1]

        return[l,b]
        

File: 492-ConstructTheRectangle/492-ConstructTheRectangle.java
// Last updated: 7/31/2025, 12:29:20 PM
class Solution {
    public int[] constructRectangle(int area) {
        int b = (int) Math.sqrt(area);
        while(area%b!=0)
           { b-=1;}
        int l=(area/b);
        if(b==1){
            return new int[]{area,1};
            }
        return new int[] {l,b};
    }
}